lib/index.js

```diff
@@ -278,6 +279,13 @@ Server.prototype = {
                                  this.userConfig.agent.handshakeTimeout);
       }
     }
+
+    if(this.userConfig.agent.mediationLayer) {
+      webAppOptions.isProxiedHttps = this.userConfig.agent.mediationLayer.enabled;
+    } else if(this.userConfig.agent.https) {
+      webAppOptions.isProxiedHttps = this.userConfig.agent.https.port > ''
+    }
+
```

Comment:
Can you rewrite lines 224 to lines 287 to be written less silly? It's becoming difficult to read and can be made a lot cleaner

i.e. 
line 224 `if (usingApiml) {
...
}`
and then line 266 follows with `if (this.userConfig.agent && !usingApiml) {` should be else if

line 283 asks for `if(this.userConfig.agent.mediationLayer) {`
yet on line 250-254 we have
`if (this.userConfig.agent
         && this.userConfig.agent.mediationLayer
         && this.userConfig.agent.mediationLayer.enabled
         && this.userConfig.agent.mediationLayer.serviceName
         && this.userConfig.node.mediationLayer.server.gatewayPort) {`
         
We are creating some weird logic gate that jumps around instead of having a flowing tree of assertions
etc.